pkg/runtime/runtime.go

```diff
@@ -630,6 +630,9 @@ func (a *DaprRuntime) buildHTTPPipeline() (httpMiddleware.Pipeline, error) {
 
 // registerPluggableComponents loads and register the loaded pluggable components.
 func (a *DaprRuntime) registerPluggableComponents() error {
```

Comment:
IMO it should never call this function if the feature is not enabled. I would not put that within this function.

Reply:
Okay, I can do that, but I chose that function because we don't have any `initRuntime` tests and I think it's a bit difficult to do because it has a lot of dependencies - and I don't believe that this is the context of this PR. Having that said, I'd rather have the test than put it inside the initRuntime, otherwise I can create an initPluggableComponents() that does this check before calling the register, at least that way it can be "testable".

wdyt @berndverst ?

Reply:
I will leave this up to dapr/dapr maintainers.

We have already proven that our feature flags work, so we don't need to specifically test for that again.

IMO your approach is less readable. If the method were called `registerPluggableComponentsIfFeatureEnabled` it makes sense - but simply `registerPluggableComponents` is weird if I am debugging a scenario where I have the feature disabled.

Thoughts?

Reply:
IMO registerPluggableComponents should only register any pluggable component that is there. It should not check if the feature is enabled or not. And having an initPluggableComponents make sense in a way to modularize the code as well as making it testable. And that initialization code checking if it has to start registering pluggable components if feature is enabled makes its more clear Imo.

Reply:
>IMO your approach is less readable. If the method were called registerPluggableComponentsIfFeatureEnabled it makes sense - but simply registerPluggableComponents is weird if I am debugging a scenario where I have the feature disabled. I would not expect that method to be entered in such a case.

I would agree with you if we had a single place where featureflags are initialized/checked, which is not the case. In that sense I prefer InitPluggableComponents because it turns the entrypoint for anyone to understand if the components will be loaded or not, even because I think the initRuntime code is getting too many lines and bad to maintain, so I think discussing name when the we have untested code ends up becoming irrelevant

Reply:
> IMO registerPluggableComponents should only register any pluggable component that is there. It should not check if the feature is enabled or not. And having an initPluggableComponents make sense in a way to modularize the code as well as making it testable. And that initialization code checking if it has to start registering pluggable components if feature is enabled makes its more clear Imo.

I agree moving to there